src/Workspaces/Remote/ServiceHub/Services/ExtensionMethodImportCompletion/RemoteExtensionMethodImportCompletionService.cs

```diff
@@ -40,9 +38,13 @@ public RemoteExtensionMethodImportCompletionService(in ServiceConstructionArgume
             bool hideAdvancedMembers,
             CancellationToken cancellationToken)
         {
+            var stopwatch = SharedStopwatch.StartNew();
             return RunServiceAsync(solutionChecksum, async solution =>
             {
-                var document = solution.GetDocument(documentId)!;
+                var assetSyncTime = stopwatch.Elapsed;
+
+                // Completion always uses frozen-partial semantic in-proc, which is not automatically passed to OOP, so enable it explicitly
+                var document = solution.GetRequiredDocument(documentId).WithFrozenPartialSemantics(cancellationToken);
```

Comment:
Ignoring the other points about this not working yet, didn't we already make a change for completion to be using frozen partials somewhere else?

Reply:
Do you mean completion in-proc? If so, then yes, we have fixed it to always use partials.